atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/DbKeyValueServiceConfig.java

```diff
@@ -29,9 +42,53 @@ public DbSharedConfig shared() {
         return ImmutableDbSharedConfig.builder().build();
     }
 
+    public abstract ConnectionConfig connection();
+
     @Value.Derived
     public abstract Supplier<DbTableFactory> tableFactorySupplier();
 
-    public abstract SqlConnectionSupplier sqlConnectionSupplier();
+    @Value.Default
+    public Supplier<SqlConnectionSupplier> sqlConnectionSupplier() {
+        return () -> createConnectionSupplier();
```

Comment:
Spoke with @joelea and I think the right path forward would be to remove the Supplier here and just have it directly return the object.  I don't see much value in the wrapped supplier for the sake of immutability